cfme/common/vm.py

```diff
@@ -1013,68 +1035,26 @@ def retire(self, entities):
         view = self.create_view(RequestsView)
         view.flash.assert_no_error()
 
-    def set_retirement_date(self, entities, when=None, offset=None, warn=None):
-        """Select and then set the retirement date for one or more VMs|Instances from
-        the All VMs|Instances page.
+    def navigate_to_set_retirement(self, **kwargs):
+        """Navigate to the Set/Remove Retirement Date form for one or more VMs/Instances
+        from the All VMs / All Instances page.
 
         Args:
-            entities: :py:class:`list` of one or more InfraVM | Instance objects. This method
-                is intended to be called from either an InfraVMCollection instance (with a list
-                of InfraVM objects, or an InstanceCollection instance (with a list of Instance
-                objects).
-            The other arguments are the same as in the :py:class:`VM` method.
+            entities: :py:class:`list` of VMs / Instances to select.
         """
-        changed = False
-        fill_date = None
-        fill_offset = None
-
-        if when and offset:
-            raise ValueError("set_retirement_date takes 'when' or 'offset', but not both")
-        if when and not isinstance(when, (datetime, date)):
-            raise ValueError("'when' argument must be a datetime object")
-
-        if offset:
-            fill_offset = {k: v
-                           for k, v in offset.items()
-                           if k in ['months', 'weeks', 'days', 'hours']}
-        elif when:
-            # format using 4-digit year for form fill
-            fill_date = when.strftime('%m/%d/%Y %H:%M')
-        else:
-            # Clear retirement date with a space in TextInput widget.
-            # Using space here because empty string does not clear date correctly
-            fill_date = ' '
+        entities = kwargs.get('entities', None)
+        assert entities, "navigate_to_set_retirement requires a list of VMs."
```

Comment:
Same here, we should really only use `assert` in test cases.

Reply:
Changed.